etcdserver/server.go

```diff
@@ -2617,6 +2678,39 @@ func (s *EtcdServer) updateClusterVersion(ver string) {
 	}
 }
 
+func (s *EtcdServer) monitorDowngrade() {
+	lg := s.getLogger()
+	for {
+		select {
+		case <-time.After(monitorDowngradeInterval):
+		case <-s.stopping:
+			return
+		}
+
+		if s.Leader() != s.ID() {
+			continue
+		}
+
+		d := s.cluster.DowngradeInfo()
+		if !d.Enabled {
+			continue
+		}
+
+		targetVersion := d.TargetVersion
+		if isDowngradeFinished(s.getLogger(), targetVersion, getVersions(s.getLogger(), s.cluster, s.id, s.peerRt)) {
+			if lg != nil {
```

Comment:
Here as well, logger nil checking should be handled at the top level. Downstream packages should not worry about this checks.